SU2_CFD/src/numerics/CNumerics.cpp

```diff
@@ -1758,40 +1756,60 @@ void CNumerics::GetPrimitive2Conservative (const su2double *val_Mean_PrimVar,
   }
 }
 
-
-
 void CNumerics::CreateBasis(const su2double *val_Normal) {
   unsigned short iDim;
   su2double modm, modl;
 
-  /*--- Define l as a vector in the plane normal to the supplied vector ---*/
-  l[0] = 0.0;
-  l[1] = -val_Normal[2];
-  l[2] = val_Normal[1];
-
-  /*--- Check for the zero vector and re-assign if needed ---*/
-  if (l[0] == 0.0 && l[1] == 0.0 && l[2] == 0.0) {
-    l[0] = -val_Normal[2];
-    l[1] = 0.0;
-    l[2] = val_Normal[0];
-  }
-
-  /*--- Take vector product of n * l to make m ---*/
-  m[0] = val_Normal[1]*l[2] - val_Normal[2]*l[1];
-  m[1] = val_Normal[2]*l[0] - val_Normal[0]*l[2];
-  m[2] = val_Normal[0]*l[1] - val_Normal[1]*l[0];
-
-  /*--- Normalize ---*/
-  modm =0 ; modl = 0;
-  for (iDim =0 ; iDim < nDim; iDim++) {
-    modm += m[iDim]*m[iDim];
-    modl += l[iDim]*l[iDim];
-  }
-  modm = sqrt(modm);
-  modl = sqrt(modl);
-  for (iDim =0 ; iDim < nDim; iDim++) {
-    l[iDim] = l[iDim]/modl;
-    m[iDim] = m[iDim]/modm;
+  if (nDim ==2){
+
+    /*--- Multiply Normal by [0 -1; 1 0] rotation matrix ---*/
+    l[0] = -val_Normal[1];
+    l[1] = val_Normal[0];
+
+    /*--- Set m matrix to zero ---*/
+    m[0] = 0.0;
+    m[1] = 0.0;
+
+    /*--- Normalize ---*/
+    modl = 0.0;
+    for (iDim = 0; iDim <nDim; iDim ++)
+      modl += l[iDim]*l[iDim];
+    modl = sqrt(modl);
+    for (iDim =0; iDim<nDim; iDim++)
+      l[iDim] = l[iDim]/modl;
+
+  } else {
+
+    /*--- Define l as a vector in the plane normal to the supplied vector ---*/
+    l[0] = 0.0;
+    l[1] = -val_Normal[2];
+    l[2] = val_Normal[1];
+
+    /*--- Check for the zero vector and re-assign if needed ---*/
+    if (l[0] == 0.0 && l[1] == 0.0 && l[2] == 0.0) {
+      l[0] = -val_Normal[2];
+      l[1] = 0.0;
+      l[2] = val_Normal[0];
+    }
+
+    /*--- Take vector product of n * l to make m ---*/
+    m[0] = val_Normal[1]*l[2] - val_Normal[2]*l[1];
+    m[1] = val_Normal[2]*l[0] - val_Normal[0]*l[2];
+    m[2] = val_Normal[0]*l[1] - val_Normal[1]*l[0];
+
+    /*--- Normalize ---*/
+    modm =0 ; modl = 0;
+    for (iDim =0 ; iDim < nDim; iDim++) {
+      modm += m[iDim]*m[iDim];
+      modl += l[iDim]*l[iDim];
+    }
+    modm = sqrt(modm);
+    modl = sqrt(modl);
```

Comment:
preemptive comment, these functions (norms, cross-products, etc.) are implemented in the GeometryToolbox.
Also as a general question for the NEMO folks, would the make sense to push these smaller fixes to develop? To avoid very large PR's, or just to avoid conflicts due to the occasional blank space fixes, formatting, etc.

Reply:
Thanks, @pcarruscag.  I think I need to generalize this CreateBasis into GetPMatrix functions.       

I agree, If the community is okay with many small PRs, I can discuss that with NEMO team.

Reply:
> I agree, If the community is okay with many small PRs

I think I remember reading it somewhere, but it is what "all" open source projects ask for.